R/cancensus.R

```diff
@@ -366,62 +378,107 @@ list_census_vectors <- function(dataset, use_cache=TRUE) {
       warning(paste("Cached vectors list may be out of date. Set `use_cache =",
                     "FALSE` to update it."))
     }
+    attributes(result)$dataset=dataset # just in case, catching cached legacy datasets
     result
   }
 }
 
+#' List all parent variables from vector hierarchichal based on a (sub-)list of census
+#' variables returned by
+#' \code{list_census_vectors} or \code{search_census_vectors}.
+#'
+#' @param vector_list The list of vectors to be used
+#'
+#' @export
+#'
+#' @examples
+#' parent_census_vectors(vector_list, 'CA16')
+parent_census_vectors <- function(vector_list){
+  base_list <- vector_list
+  dataset=attributes(base_list)$dataset
+  n=0
+  vector_list <- list_census_vectors(dataset) %>%
+    filter(vector %in% base_list$parent_vector) %>%
+    distinct(vector, .keep_all = TRUE)
+  while (n!=nrow(vector_list)) {
+    n=nrow(vector_list)
+    new_list <- list_census_vectors(dataset) %>%
+      filter(vector %in% vector_list$parent_vector)
+    vector_list <- vector_list %>% rbind(new_list) %>%
+      distinct(vector, .keep_all = TRUE)
+  }
+  attributes(vector_list)$dataset=dataset
+  return(vector_list)
+}
+
+#' List all child variables from vector hierarchichal based on a (sub-)list of census
+#' variables returned by
+#' \code{list_census_vectors} or \code{search_census_vectors}.
+#'
+#' @param vector_list The list of vectors to be used
+#' @param leaves_only Boolean flag to indicate if only leaf vectors should be returned,
+#' i.e. vectors that don't have children
+#'
+#' @export
+#'
+#' @examples
+#' child_census_vectors(vector_list, 'CA16')
+child_census_vectors <- function(vector_list, leaves_only=FALSE){
+  dataset=attributes(base_list)$dataset
+  base_list <- vector_list
+  n=0
+  vector_list <- list_census_vectors(dataset) %>%
+    filter(parent_vector %in% base_list$vector) %>%
```

Comment:
This `filter` (and many other functions in this PR) should be prefixed with `dplyr::`.